lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,679 @@
+        private static TaskManager TaskManagerWithTasksRunning()
+        {
+            var taskManager = TaskManagerWithTasksSubmitted();
+            taskManager.RecordRunningTask(CreateMockRunningTask(MasterTaskId));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 2));
+
+            return taskManager;
+        }
+
+        /// <summary>
+        /// Creates a mock IActiveContext
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        private static IActiveContext CreateMockActiveContext(int id)
+        {
+            var mockActiveContext = Substitute.For<IActiveContext>();
+            mockActiveContext.Id.Returns(ContextIdPrefix + id);
+            mockActiveContext.EvaluatorId.Returns(EvaluatorIdPrefix + ContextIdPrefix + id);
+            return mockActiveContext;
+        }
+
+        /// <summary>
+        /// Creates a mock FailedTask with specified taskId and error message
+        /// </summary>
+        /// <param name="taskId"></param>
+        /// <param name="errorMsg"></param>
+        /// <returns></returns>
+        private static IFailedTask CreateMockFailedTask(string taskId, string errorMsg)
+        {
+            IFailedTask failedtask = Substitute.For<IFailedTask>();
+            failedtask.Id.Returns(taskId);
+            failedtask.Message.Returns(errorMsg);
+            return failedtask;
+        }
+
+        /// <summary>
+        /// Creates a mock running task with the taskId specified
+        /// </summary>
+        /// <param name="taskId"></param>
+        /// <returns></returns>
+        private static IRunningTask CreateMockRunningTask(string taskId)
+        {
+            var runningTask = Substitute.For<IRunningTask>();
+            runningTask.Id.Returns(taskId);
+            return runningTask;
+        }
+
+        /// <summary>
+        /// Creates a mock running task with the taskId specified
+        /// </summary>
+        /// <param name="taskId"></param>
+        /// <returns></returns>
+        private static ICompletedTask CreateMockCompletedTask(string taskId)
+        {
+            var completedTask = Substitute.For<ICompletedTask>();
+            completedTask.Id.Returns(taskId);
+            return completedTask;
+        }
+
+        /// <summary>
+        /// Creates a mock IFailedEvaluator with the specified IFailedTask associated
+        /// </summary>
+        /// <param name="evaluatorId"></param>
+        /// <param name="failedTask"></param>
+        /// <returns></returns>
+        private static IFailedEvaluator CreateMockFailedEvaluator(string evaluatorId, IFailedTask failedTask)
+        {
+            var failedEvalutor = Substitute.For<IFailedEvaluator>();
+            failedEvalutor.Id.Returns(evaluatorId);
+            failedEvalutor.FailedTask.Returns(Optional<IFailedTask>.Of(failedTask));
```

Comment:
Ohh yesterday I was reviewing from my laptop and indentations were all screwed up...Looks fine from my desktop.